File: trunk/main/src/com/seovic/core/extractor/KeyExtractorAdapter.java
/*
 * Copyright 2009 Aleksandar Seovic
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.seovic.core.extractor;


import com.tangosol.util.ValueExtractor;
import com.tangosol.util.extractor.KeyExtractor;

import com.tangosol.io.pof.PortableObject;

import com.seovic.core.Extractor;

import java.io.Serializable;


/**
 * Adapter that allows any class that implements <tt>com.tangosol.util.ValueExtractor</tt>
 * to be used where {@link Extractor} instance is expected.
 *
 * @author Aleksandar Seovic  2009.09.20
 */
public class KeyExtractorAdapter
    extends KeyExtractor
    implements Extractor, Serializable, PortableObject
    {
    // ---- constructors ----------------------------------------------------

    /**
     * Deserialization constructor (for internal use only).
     */
    public KeyExtractorAdapter()
        {
        }

    /**
     * Construct a <tt>KeyExtractorAdapter</tt> instance.
     *
     * @param delegate  value extractor to delegate to
     */
    public KeyExtractorAdapter(ValueExtractor delegate)
        {
        super(delegate);
        }
    }

File: main.net/src/Coherence.Tools/Coherence/Util/Extractor/KeyExtractorAdapter.cs
using System;

using Seovic.Core;

using Tangosol.Util;
using Tangosol.Util.Extractor;

namespace Seovic.Coherence.Util.Extractor
{
    /// <summary>
    /// Adapter that allows any class that implements <c>Tangosol.Util.IValueExtractor</c>
    /// to be used where <see cref="IExtractor"/> instance is expected.
    /// </summary>
    /// <author>Ivan Cikic  2010.02.05</author>
    /// <author>Aleksandar Seovic  2010.02.05</author>
    [Serializable]
    public class KeyExtractorAdapter : KeyExtractor, IExtractor
    {
        #region Constructor

        /// <summary>
        /// Deserialization constructor (for internal use only).
        /// </summary>
        public KeyExtractorAdapter()
        {
        }

        /// <summary>
        /// Construct a <code>ValueExtractorAdapter</code> instance.
        /// </summary>
        /// <param name="extractor">Value extractor to delegate to.</param>
        public KeyExtractorAdapter(IValueExtractor extractor)
            : base(extractor)
        {
        }

        #endregion
    }
}